soccer/gameplay/tactics/defense.py

```diff
@@ -48,9 +48,11 @@ def __init__(self, defender_priorities=[20, 19]):
                             lambda: not self.should_clear_ball(),
                             "done clearing")
 
-        goalie = tactics.positions.submissive_goalie.SubmissiveGoalie()
-        goalie.shell_id = main.root_play().goalie_id
-        self.add_subbehavior(goalie, "goalie", required=False)
+        valid_shells = [bot.shell_id for bot in main.our_robots()]
+        if main.root_play().goalie_id in valid_shells:
```

Comment:
Keep in mind that since this happens at `__init__`, a goalie won't be assigned if the right robot shows up later on.